diff --git a/grub-core/commands/nativedisk.c b/grub-core/commands/nativedisk.c
--- a/grub-core/commands/nativedisk.c
+++ b/grub-core/commands/nativedisk.c
@@ -89,6 +89,8 @@ get_uuid (const char *name, char **uuid, int getnative)
     case GRUB_DISK_DEVICE_HOST_ID:
       /* To access coreboot roms.  */
     case GRUB_DISK_DEVICE_CBFSDISK_ID:
+      /* Virtual file system */
+    case GRUB_DISK_DEVICE_VFS_ID:
       /* GRUB-only memdisk. Can't match any of firmware devices.  */
     case GRUB_DISK_DEVICE_MEMDISK_ID:
       grub_dprintf ("nativedisk", "Skipping native disk %s\n",
diff --git a/include/grub/disk.h b/include/grub/disk.h
--- a/include/grub/disk.h
+++ b/include/grub/disk.h
@@ -50,7 +50,8 @@ enum grub_disk_dev_id
     GRUB_DISK_DEVICE_UBOOTDISK_ID,
     GRUB_DISK_DEVICE_XEN,
     GRUB_DISK_DEVICE_OBDISK_ID,
-    GRUB_DISK_DEVICE_VHD_ID
+    GRUB_DISK_DEVICE_VHD_ID,
+    GRUB_DISK_DEVICE_VFS_ID
   };
 
 struct grub_disk;
